avocado/utils/memory.py

```diff
@@ -216,6 +216,25 @@ def numa_nodes():
     return (sorted(nodes))
 
 
+def numa_nodes_with_memory():
+    """
+    Get a list of NUMA nodes present with memory on the system.
+
+    :return: List with nodes which has memory.
+    """
+    nodes = []
+    mem_path = '/sys/devices/system/node/has_normal_memory'
+    if not os.path.exists(mem_path):
+        mem_path = '/sys/devices/system/node/has_memory'
+    node_list = str(genio.read_file(mem_path).rstrip('\n')).split(',')
+    for nodeset in node_list:
+        if '-' in nodeset:
+            nodes.extend([int(node) for node in nodeset.split('-')])
```

Comment:
I think you lack a `range` here, otherwise you only include the boundary ones.

Reply:
Actually this function might be quite useful. After fixing it please consider adding it to `avocado.utils.data_structures`. I'd call it `comma_separated_ranges_to_list` but it's just first guess, feel free to come-up with a better name.